File: packages/soulseek_protocol/lib/src/messages/codes.dart
// Soulseek message codes
// Based on Nicotine+ SLSKPROTOCOL.md specification

class ServerCode {
  ServerCode._();

  static const int login = 1;
  static const int setListenPort = 2;
  static const int getPeerAddress = 3;
  static const int addUser = 5;
  static const int parentInactivityTimeout = 8;
  static const int parentMinSpeed = 9;
  static const int searchRequest = 26;
  static const int searchResponse = 28;
  static const int userExists = 34;
  static const int userExistsResponse = 35;
  static const int getStatus = 36;
  static const int statusResponse = 37;
  static const int ping = 40;
  static const int pong = 41;
  static const int joinRoom = 43;
  static const int leaveRoom = 44;
  static const int userJoinedRoom = 45;
  static const int userLeftRoom = 46;
  static const int roomMessage = 47;
  static const int roomList = 48;
  static const int privateMessage = 51;
  static const int privateMessageACK = 52;
  static const int acknowledgeNotifyACL = 53;
  static const int notifyACL = 54;
  static const int serverShuttingDown = 55;
  static const int checkPrivileges = 56;
  static const int privilegesResponse = 57;
  static const int checkDownloadQueue = 58;
  static const int checkDownloadQueueResp = 59;
  static const int userSearchResponse = 60;
  static const int addPrivilegedUser = 61;
  static const int removePrivilegedUser = 62;
  static const int acceptPrivilegedUser = 63;
  static const int placeInQueueResponse = 64;
  static const int parentPathMerge = 65;
  static const int parentPathAlias = 66;
  static const int wishes = 67;
  static const int wishReply = 68;
  static const int wishlistInclusion = 69;
  static const int configuration = 100;
  static const int distributedAliveInterval = 102;
  static const int addThing = 110;
  static const int removeThing = 111;
  static const int thingsAtPlace = 112;
  static const int thingsAtPlaceReply = 113;
  static const int thingGroups = 114;
  static const int thingGroupsReply = 115;
  static const int placeGroups = 116;
  static const int placeGroupsReply = 117;
  static const int fileSearch = 120;
  static const int fileSearchReply = 121;
  static const int suggestSimilar = 122;
  static const int similarUsers = 123;
  static const int similarUsersReply = 124;
  static const int comments = 125;
  static const int commentsReply = 126;
  static const int itemFolder = 127;
  static const int itemFolderReply = 128;
  static const int itemPath = 129;
  static const int itemPathReply = 130;
  static const int folderContents = 131;
  static const int folderContentsReply = 132;
  static const int userPrivileges = 133;
  static const int userPrivilegesReply = 134;
  static const int privateRoomUsers = 135;
  static const int privateRoomUsersReply = 136;
  static const int privateRoomAddUser = 137;
  static const int privateRoomRemoveUser = 138;
  static const int privateRoomDismember = 139;
  static const int privateRoomToggle = 140;
  static const int privateRoomAdded = 141;
  static const int privateRoomRemoved = 142;
  static const int privateRoomDismembered = 143;
  static const int privateRoomOwners = 144;
  static const int privateRoomOwnersReply = 145;
  static const int similarEntry = 146;
  static const int similarEntryReply = 147;
  static const int roomTickers = 148;
  static const int roomTickersReply = 149;
  static const int roomTickerSet = 150;
  static const int roomTickerRemove = 151;
  static const int roomSearch = 152;
  static const int sendUploadSpeed = 153;
  static const int userStats = 154;
  static const int userStatsReply = 155;
  static const int privateRoomAddPrivilegedUser = 156;
  static const int privateRoomRemovePrivilegedUser = 157;
  static const int privateRoomACL = 158;
  static const int privateRoomACLReply = 159;
  static const int accountNotify = 160;
  static const int embeddedMessage = 161;
  static const int acceptChildren = 162;
  static const int interestAdd = 163;
  static const int interestRemove = 164;
  static const int interestReply = 165;
  static const int adminMessage = 166;
  static const int globalMessage = 167;
  static const int recommendedFileSearch = 168;
  static const int searchRequestSocial = 169;
  static const int searchParent = 170;
  static const int searchChild = 171;
  static const int killParentSearch = 172;
  static const int killChildSearch = 173;
  static const int addDistributedChild = 174;
  static const int removeDistributedChild = 175;
  static const int addDistributedParent = 176;
  static const int removeDistributedParent = 177;
  static const int branchLevel = 178;
  static const int branchRoot = 179;
  static const int childDepth = 180;
  static const int distributedSearch = 181;
  static const int compressedSearch = 182;
  static const int degradedSearch = 183;
  static const int loginResponse = 206;
  static const int changePassword = 207;
  static const int newPassword = 208;
  static const int kickMessage = 209;
  static const int banMessage = 210;
  static const int setListenPort2 = 211;
  static const int messageUsers = 212;
  static const int messageUsersReply = 213;
  static const int privateRoomSearch = 214;
  static const int privateRoomSearchReply = 215;
}

class PeerCode {
  PeerCode._();

  static const int fileSearch = 120;
  static const int fileSearchReply = 121;
  static const int folderContents = 131;
  static const int folderContentsReply = 132;
  static const int userInfo = 133;
  static const int userInfoReply = 134;
  static const int transferRequest = 135;
  static const int transferResponse = 136;
  static const int placehold = 137;
  static const int queueFailed = 138;
  static const int placeInQueue = 139;
  static const int uploadFailed = 140;
  static const int queueUpload = 141;
  static const int placeInQueueResponse = 142;
  static const int uploadDenied = 143;
  static const int userPrivileges = 144;
  static const int userPrivilegesReply = 145;
  static const int searchResponse = 147;
  static const int infoRequest = 148;
  static const int infoReply = 149;
  static const int folderSearch = 150;
  static const int folderSearchReply = 151;
}

class DistantCode {
  DistantCode._();

  static const int fileSearch = 3;
  static const int fileSearchReply = 4;
}

class Direction {
  Direction._();

  static const int download = 0;
  static const int upload = 1;
}

File: Soulseek.Protocol/Messages/Codes.cs
namespace Soulseek.Protocol.Messages;

public static class ServerCode
{
    public const int Login = 1;
    public const int SetListenPort = 2;
    public const int GetPeerAddress = 3;
    public const int AddUser = 5;
    public const int ParentInactivityTimeout = 8;
    public const int ParentMinSpeed = 9;
    public const int SearchRequest = 26;
    public const int SearchResponse = 28;
    public const int UserExists = 34;
    public const int UserExistsResponse = 35;
    public const int GetStatus = 36;
    public const int StatusResponse = 37;
    public const int Ping = 40;
    public const int Pong = 41;
    public const int JoinRoom = 43;
    public const int LeaveRoom = 44;
    public const int UserJoinedRoom = 45;
    public const int UserLeftRoom = 46;
    public const int RoomMessage = 47;
    public const int RoomList = 48;
    public const int PrivateMessage = 51;
    public const int PrivateMessageAck = 52;
    public const int AcknowledgeNotifyAcl = 53;
    public const int NotifyAcl = 54;
    public const int ServerShuttingDown = 55;
    public const int CheckPrivileges = 56;
    public const int PrivilegesResponse = 57;
    public const int CheckDownloadQueue = 58;
    public const int CheckDownloadQueueResp = 59;
    public const int UserSearchResponse = 60;
    public const int AddPrivilegedUser = 61;
    public const int RemovePrivilegedUser = 62;
    public const int AcceptPrivilegedUser = 63;
    public const int PlaceInQueueResponse = 64;
    public const int ParentPathMerge = 65;
    public const int ParentPathAlias = 66;
    public const int Wishes = 67;
    public const int WishReply = 68;
    public const int WishlistInclusion = 69;
    public const int Configuration = 100;
    public const int DistributedAliveInterval = 102;
    public const int AddThing = 110;
    public const int RemoveThing = 111;
    public const int ThingsAtPlace = 112;
    public const int ThingsAtPlaceReply = 113;
    public const int ThingGroups = 114;
    public const int ThingGroupsReply = 115;
    public const int PlaceGroups = 116;
    public const int PlaceGroupsReply = 117;
    public const int FileSearch = 120;
    public const int FileSearchReply = 121;
    public const int SuggestSimilar = 122;
    public const int SimilarUsers = 123;
    public const int SimilarUsersReply = 124;
    public const int Comments = 125;
    public const int CommentsReply = 126;
    public const int ItemFolder = 127;
    public const int ItemFolderReply = 128;
    public const int ItemPath = 129;
    public const int ItemPathReply = 130;
    public const int FolderContents = 131;
    public const int FolderContentsReply = 132;
    public const int UserPrivileges = 133;
    public const int UserPrivilegesReply = 134;
    public const int PrivateRoomUsers = 135;
    public const int PrivateRoomUsersReply = 136;
    public const int PrivateRoomAddUser = 137;
    public const int PrivateRoomRemoveUser = 138;
    public const int PrivateRoomDismember = 139;
    public const int PrivateRoomToggle = 140;
    public const int PrivateRoomAdded = 141;
    public const int PrivateRoomRemoved = 142;
    public const int PrivateRoomDismembered = 143;
    public const int PrivateRoomOwners = 144;
    public const int PrivateRoomOwnersReply = 145;
    public const int SimilarEntry = 146;
    public const int SimilarEntryReply = 147;
    public const int RoomTickers = 148;
    public const int RoomTickersReply = 149;
    public const int RoomTickerSet = 150;
    public const int RoomTickerRemove = 151;
    public const int RoomSearch = 152;
    public const int SendUploadSpeed = 153;
    public const int UserStats = 154;
    public const int UserStatsReply = 155;
    public const int PrivateRoomAddPrivilegedUser = 156;
    public const int PrivateRoomRemovePrivilegedUser = 157;
    public const int PrivateRoomAcl = 158;
    public const int PrivateRoomAclReply = 159;
    public const int AccountNotify = 160;
    public const int EmbeddedMessage = 161;
    public const int AcceptChildren = 162;
    public const int InterestAdd = 163;
    public const int InterestRemove = 164;
    public const int InterestReply = 165;
    public const int AdminMessage = 166;
    public const int GlobalMessage = 167;
    public const int RecommendedFileSearch = 168;
    public const int SearchRequestSocial = 169;
    public const int SearchParent = 170;
    public const int SearchChild = 171;
    public const int KillParentSearch = 172;
    public const int KillChildSearch = 173;
    public const int AddDistributedChild = 174;
    public const int RemoveDistributedChild = 175;
    public const int AddDistributedParent = 176;
    public const int RemoveDistributedParent = 177;
    public const int BranchLevel = 178;
    public const int BranchRoot = 179;
    public const int ChildDepth = 180;
    public const int DistributedSearch = 181;
    public const int CompressedSearch = 182;
    public const int DegradedSearch = 183;
    public const int LoginResponse = 206;
    public const int ChangePassword = 207;
    public const int NewPassword = 208;
    public const int KickMessage = 209;
    public const int BanMessage = 210;
    public const int SetListenPort2 = 211;
    public const int MessageUsers = 212;
    public const int MessageUsersReply = 213;
    public const int PrivateRoomSearch = 214;
    public const int PrivateRoomSearchReply = 215;
}

public static class PeerCode
{
    public const int FileSearch = 120;
    public const int FileSearchReply = 121;
    public const int FolderContents = 131;
    public const int FolderContentsReply = 132;
    public const int UserInfo = 133;
    public const int UserInfoReply = 134;
    public const int TransferRequest = 135;
    public const int TransferResponse = 136;
    public const int Placehold = 137;
    public const int QueueFailed = 138;
    public const int PlaceInQueue = 139;
    public const int UploadFailed = 140;
    public const int QueueUpload = 141;
    public const int PlaceInQueueResponse = 142;
    public const int UploadDenied = 143;
    public const int UserPrivileges = 144;
    public const int UserPrivilegesReply = 145;
    public const int SearchResponse = 147;
    public const int InfoRequest = 148;
    public const int InfoReply = 149;
    public const int FolderSearch = 150;
    public const int FolderSearchReply = 151;
}

public static class DistantCode
{
    public const int FileSearch = 3;
    public const int FileSearchReply = 4;
}

public static class Direction
{
    public const int Download = 0;
    public const int Upload = 1;
}